src/Features/VisualBasic/Portable/ExtractInterface/VisualBasicExtractInterfaceService.vb

```diff
@@ -88,46 +71,39 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.ExtractInterface
         End Function
 
-        Private Function UpdateTypeWithImplementsClause(
-            solutionWithInterfaceDocument As Solution,
-            invocationLocationDocument As DocumentId,
-            typeNodeAnnotation As SyntaxAnnotation,
-            extractedInterfaceSymbol As INamedTypeSymbol,
-            docToRootMap As Dictionary(Of DocumentId, CompilationUnitSyntax),
-            cancellationToken As CancellationToken) As String
+        Private Function GetUpdatedImplementsClause(implementsClause As ImplementsClauseSyntax, qualifiedName As QualifiedNameSyntax) As ImplementsClauseSyntax
+            If implementsClause IsNot Nothing Then
+                Return implementsClause.AddInterfaceMembers(qualifiedName).WithAdditionalAnnotations(Formatter.Annotation)
+            Else
+                Return SyntaxFactory.ImplementsClause(qualifiedName).WithAdditionalAnnotations(Formatter.Annotation)
+            End If
+        End Function
+
+        Private Function CreateFinalSolution(solutionWithInterfaceDocument As Solution, documentIds As IEnumerable(Of DocumentId), docToRootMap As Dictionary(Of DocumentId, CompilationUnitSyntax)) As Solution
+            Dim finalSolution = solutionWithInterfaceDocument
+
+            For Each docId In documentIds
+                finalSolution = finalSolution.WithDocumentSyntaxRoot(docId, docToRootMap(docId), PreservationMode.PreserveIdentity)
+            Next
+
+            Return finalSolution
+        End Function
+
+        Friend Overrides Function ShouldIncludeAccessibilityModifier(typeNode As SyntaxNode) As Boolean
+            Dim typeDeclaration = DirectCast(typeNode, TypeBlockSyntax)
+            Return typeDeclaration.GetModifiers().Any(Function(m) SyntaxFacts.IsAccessibilityModifier(m.Kind()))
+        End Function
 
-            Dim documentWithTypeNode = solutionWithInterfaceDocument.GetDocument(invocationLocationDocument)
-            Dim typeDeclaration = documentWithTypeNode.GetSyntaxRootSynchronously(cancellationToken).GetAnnotatedNodes(Of TypeBlockSyntax)(typeNodeAnnotation).Single()
+        Friend Overrides Function UpdateMembersWithExplicitImplementations(unformattedSolution As Solution, documentIds As IReadOnlyList(Of DocumentId), extractedInterfaceSymbol As INamedTypeSymbol, typeToExtractFrom As INamedTypeSymbol, includedMembers As IEnumerable(Of ISymbol), symbolToDeclarationAnnotationMap As Dictionary(Of ISymbol, SyntaxAnnotation), cancellationToken As CancellationToken) As Solution
+            Dim docToRootMap = New Dictionary(Of DocumentId, CompilationUnitSyntax)
 
             Dim implementedInterfaceStatementSyntax = If(extractedInterfaceSymbol.TypeParameters.Any(),
                 SyntaxFactory.GenericName(
                     SyntaxFactory.Identifier(extractedInterfaceSymbol.Name),
                     SyntaxFactory.TypeArgumentList(SyntaxFactory.SeparatedList(extractedInterfaceSymbol.TypeParameters.Select(Function(p) SyntaxFactory.ParseTypeName(p.Name))))),
```

Comment:
same for parseTypeName. it is TypeParmater so it will be like T or things like that?

Reply:
I didn't change this code, but let me make sure we have this covered: I believe you're just talking about generics right? That's tested with http://source.roslyn.io/#Microsoft.CodeAnalysis.VisualBasic.EditorFeatures.UnitTests/ExtractInterface/ExtractInterfaceTests.vb,f8749251ffda727e 

Does that test cover the scenario you are talking about?